mailbox/api/src/main/java/org/apache/james/mailbox/MessageSequenceNumber.java

```diff
@@ -0,0 +1,64 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import com.google.common.base.Preconditions;
+
+public final class MessageSequenceNumber {
+
+    public static MessageSequenceNumber of(int msn) {
+        Preconditions.checkArgument(msn >= 0);
+        return new MessageSequenceNumber(msn);
+    }
+
+    private final int msn;
+
+    private MessageSequenceNumber(int msn) {
+        this.msn = msn;
+    }
+
+    public int asInt() {
+        return msn;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof MessageSequenceNumber) {
+            MessageSequenceNumber that = (MessageSequenceNumber) o;
+            return Objects.equals(msn, that.msn);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(msn);
+    }
+
+    @Override
+    public String toString() {
+        return new StringJoiner(", ", MessageSequenceNumber.class.getSimpleName() + "[", "]")
```

Comment:
why not using toStringHelper?

Reply:
new IDE setup that is not properly configured, sorry